src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -671,18 +671,22 @@ bool is_additional_superset(
     //  schema  |   true   |  recurse with {}
     //  schema  |   false  |  recurse with {"not":{}}
 
-    auto field_name = [&] {
+    // in draft 2020, if checking "type": "array", "items" is used to represent
+    // additional tuples items instead of an "additionalItems"
+    auto get_field_name = [&](json_schema_dialect d) {
         switch (field_type) {
         case additional_field_for::object:
             return "additionalProperties";
         case additional_field_for::array:
-            return "additionalItems";
+            return d == json_schema_dialect::draft202012 ? "items"
```

Comment:
nit: should this be `>=` to be future-proof? (nit because these are equivalent for now)
```suggestion
            return d >= json_schema_dialect::draft202012 ? "items"
```

Reply:
yeah seems sensible. i'll add a fix in the next pr

Reply:
@andijcr was this addressed in a new pr?

Reply:
will. currently here https://github.com/redpanda-data/redpanda/compare/dev...andijcr:redpanda:fix/core-7037/schema_registry_bugs